js/views/userPageVw.js

```diff
@@ -543,25 +542,17 @@ UserPageVw = pageVw.extend({
     }
 
     if (state == "listing" || state == "listingOld" || state == "listingNew") {
-      isItemType = true;
-    }
-
-    //set address bar
-    if (isItemType) {
       addressState = "/listing";
+      addressState = hash ? addressState + "/" + hash : addressState;
     } else {
       addressState = "/" + state;
     }
-    currentAddress = this.model.get('page').profile.guid + addressState;
-    currentHandle = currentHandle ? currentHandle + addressState : "";
-    if (isItemType && hash) {
-      currentAddress += "/"+ hash;
-      currentHandle = currentHandle ? currentHandle += "/"+ hash : "";
-    } else if (addressState === "createStore"){
-      currentAddress = this.model.get('page').profile.guid;
-    }
 
-    window.obEventBus.trigger("setAddressBar", {'addressText': currentAddress, 'handle': currentHandle});
+    currentAddress = currentHandle || this.model.get('page').profile.guid;
+    currentAddress += addressState;
+
+    window.obEventBus.trigger("setAddressBar", {'addressText': currentAddress});
+    currentHandle && app.appBar.setTitle(currentHandle);
```

Comment:
Why is the title being set in setState? Doesn't this need to be set once per view load? Seems like this could be set once after the profile model is fetched.